README.md

```diff
@@ -306,6 +306,510 @@ val byteToHex = FunSpec.builder("byteToHex")
+    .addType(companion)
+    .build()
+```
+
+You can provide an optional name for a companion object.
+
+### Enums
+
+Use `enumBuilder` to create the enum type, and `addEnumConstant()` for each value:
+
+```kotlin
+val helloWorld = TypeSpec.enumBuilder("Roshambo")
+    .addEnumConstant("ROCK")
+    .addEnumConstant("SCISSORS")
+    .addEnumConstant("PAPER")
+    .build()
+```
+
+To generate this:
+
+```kotlin
+enum class Roshambo {
+    ROCK,
+
+    SCISSORS,
+
+    PAPER
+}
+```
+
+Fancy enums are supported, where the enum values override methods or call a superclass constructor.
+Here's a comprehensive example:
+
+```kotlin
+val helloWorld = TypeSpec.enumBuilder("Roshambo")
+    .primaryConstructor(FunSpec.constructorBuilder()
+        .addParameter("handsign", String::class)
+        .build())
+    .addEnumConstant("ROCK", TypeSpec.anonymousClassBuilder()
+        .addSuperclassConstructorParameter("%S", "fist")
+        .addFunction(FunSpec.builder("toString")
+            .addModifiers(KModifier.OVERRIDE)
+            .addStatement("return %S", "avalanche!")
+            .returns(String::class)
+            .build())
+        .build())
+    .addEnumConstant("SCISSORS", TypeSpec.anonymousClassBuilder()
+        .addSuperclassConstructorParameter("%S", "peace")
+        .build())
+    .addEnumConstant("PAPER", TypeSpec.anonymousClassBuilder()
+        .addSuperclassConstructorParameter("%S", "flat")
+        .build())
+    .addProperty(PropertySpec.builder("handsign", String::class, KModifier.PRIVATE)
+        .initializer("handsign")
+        .build())
+    .build()
+```
+
+Which generates this:
+
+```kotlin
+enum class Roshambo(private val handsign: kotlin.String) {
+    ROCK("fist") {
+        override fun toString(): kotlin.String = "avalanche!"
+    },
+
+    SCISSORS("peace"),
+
+    PAPER("flat");
+}
+```
+
+### Anonymous Inner Classes
+
+In the enum code, we used `TypeSpec.anonymousClassBuilder()`. Anonymous inner classes can also be 
+used in code blocks. They are values that can be referenced with `%L`:
+
+```kotlin
+val comparator = TypeSpec.anonymousClassBuilder()
+    .addSuperinterface(Comparator::class.parameterizedBy(String::class))
+    .addFunction(FunSpec.builder("compare")
+        .addModifiers(KModifier.OVERRIDE)
+        .addParameter("a", String::class)
+        .addParameter("b", String::class)
+        .returns(Int::class)
+        .addStatement("return %N.length - %N.length", "a", "b")
+        .build())
+    .build()
+
+val helloWorld = TypeSpec.classBuilder("HelloWorld")
+    .addFunction(FunSpec.builder("sortByLength")
+        .addParameter("strings", List::class.parameterizedBy(String::class))
+        .addStatement("%N.sortedWith(%L)", "strings", comparator)
+        .build())
+    .build()
+```
+
+This generates a method that contains a class that contains a method:
+
+```kotlin
+class HelloWorld {
+    fun sortByLength(strings: kotlin.collections.List<kotlin.String>) {
```

Comment:
are our imports broken?

Reply:
No, this output is produced by `TypeSpec.toString()`, rather than `FileSpec.toString()`. I can make a pass and strip all imports for better readability.